server/server_test.go

```diff
@@ -0,0 +1,422 @@
+	}
+	if res == nil {
+		t.Fatal("Empty result")
+	}
+	if res.ParsedExpr == nil {
+		t.Fatal("Empty parsed expression in result")
+	}
+	// Could check against 'parsed' above,
+	// but the expression ids are arbitrary,
+	// and explicit comparison logic is about as
+	// much work as normalization would be.
+	if res.ParsedExpr.Expr == nil {
+		t.Fatal("Empty expression in result")
+	}
+	switch res.ParsedExpr.Expr.ExprKind.(type) {
+	case *syntax.Expr_CallExpr:
+		c := res.ParsedExpr.Expr.GetCallExpr()
+		if c.Target != nil {
+			t.Error("Call has target", c)
+		}
+		if c.Function != "_+_" {
+			t.Error("Wrong function", c)
+		}
+		if len(c.Args) != 2 {
+			t.Error("Too many or few args", c)
+		}
+		for i, a := range c.Args {
+			switch a.ExprKind.(type) {
+			case *syntax.Expr_LiteralExpr:
+				l := a.GetLiteralExpr()
+				switch l.LiteralKind.(type) {
+				case *syntax.Literal_Int64Value:
+					if l.GetInt64Value() != int64(1) {
+						t.Errorf("Arg %d wrong value: %v", i, a)
+					}
+				default:
+					t.Errorf("Arg %d not int: %v", i, a)
+				}
+			default:
+				t.Errorf("Arg %d not literal: %v", i, a)
+			}
+		}
+	default:
+		t.Error("Wrong expression type", res.ParsedExpr.Expr)
+	}
+}
+
+func TestCheck(t *testing.T) {
+	// If TestParse() passes, it validates a good chunk
+	// of the server mechanisms for data conversion, so we
+	// won't be as fussy here..
+	req := cel_service.CheckRequest{
+		ParsedExpr: parsed,
+	}
+	res, err := globals.client.Check(context.Background(), &req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if res == nil {
+		t.Fatal("Empty result")
+	}
+	if res.CheckedExpr == nil {
+		t.Fatal("No checked expression")
+	}
+	tp, present := res.CheckedExpr.TypeMap[int64(1)]
+	if !present {
+		t.Fatal("No type for top level expression", res)
+	}
+	switch tp.TypeKind.(type) {
+	case *checked.Type_Primitive:
+		if tp.GetPrimitive() != checked.Type_INT64 {
+			t.Error("Bad top-level type", tp)
+		}
+	default:
+		t.Error("Bad top-level type", tp)
+	}
+}
+
+func TestEval(t *testing.T) {
+	req := cel_service.EvalRequest{
+		ExprKind: &cel_service.EvalRequest_ParsedExpr{parsed},
+	}
+	res, err := globals.client.Eval(context.Background(), &req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if res == nil || res.Result == nil {
+		t.Fatal("Nil result")
+	}
+	switch res.Result.Kind.(type) {
+	case *eval.ExprValue_Value:
+		v := res.Result.GetValue()
+		switch v.Kind.(type) {
+		case *value.Value_Int64Value:
+			if v.GetInt64Value() != int64(2) {
+				t.Error("Wrong result for 1 + 1", v)
+			}
+		default:
+			t.Error("Wrong result value type", v)
+		}
+	default:
+		t.Fatal("Result not a value", res.Result)
+	}
+}
+
+func TestFullUp(t *testing.T) {
+	preq := cel_service.ParseRequest{
+		CelSource: "x + y",
+	}
+	pres, err := globals.client.Parse(context.Background(), &preq)
+	if err != nil {
+		t.Fatal(err)
+	}
+	parsedExpr := pres.ParsedExpr
+	if parsedExpr == nil {
+		t.Fatal("Empty parsed expression")
+	}
+
+	intDecl := checked.Decl_Ident{
```

Comment:
Consider using `checker/decls/decls.go` helper routines to simplify the proto construction here.

Reply:
Done.  Pretty!